model/scope.go

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Scopes []Scope
+type Scope string
+
+const (
+	ScopeChannelJoin  Scope = "channels:join"
+	ScopeMessageRead  Scope = "message:read"
+	ScopeMessageWrite Scope = "message:write"
+	ScopeFilesRead    Scope = "files:read"
+	ScopeFilesWrite   Scope = "files:write"
+	ScopeSearchPosts  Scope = "search:post"
+	ScopeNoScope      Scope = ""
```

Comment:
2/5 rename `ScopeNoScope` to `AnyScope`, more intention-revealing and less typographically redundant.

Reply:
My problem with this renaming would be that we would have in the same namespace the constant "ScopeAny" (to define the any scope on API level), and the function "ScopeAny" (to create a Scope[] from a list of scopes).

I am open to change both names, but not sure which name to choose.

Reply:
Then how about `NoScope`? Seems unique and self-explanatory?